[rayyan35p] iP

rayyan35p opened the pull request:
# ChatBob 👷‍♂️ 
> _Can we fix it? Yes, we can!_
- Did some work but made mistakes?
- Think you can't fix your mistakes?
- Want to be told that you **CAN** fix it?
- If you answered yes to any of the above, you're in the right place!

## Bob is here to help!
### Bob can:

1. Keep track of your tasks
2. And if you ask `canWeFixIt()`, he will return true.

## Bob worked really hard on this...
### Look at all this code he wrote!

```java
public String getFirstInput() {
        System.out.println("Hello! I'm Bob");
        System.out.println("What can I do for you?");
        return this.getInput();
    }
```

## Bob still can't do everything yet...
> Still under construction!
- [x] Keep track of your tasks
- [ ] Tell you that you can fix it

So in the meantime, watch [this ](https://www.youtube.com/watch?v=-P2-Nmk1uFQ)while waiting for construction to complete!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

rayyan35p committed:
Replace initial code with new skeletal version

rayyan35p committed:
Modify chatbot to echo user commands

rayyan35p committed:
Add ability to store tasks in a list and print the list

rayyan35p committed:
Add Task class

rayyan35p committed:
Update Task class

rayyan35p committed:
Add TaskList class

rayyan35p committed:
Integrate Task and TaskList into main code

rayyan35p committed:
Add Deadline, Event, ToDo classes

rayyan35p committed:
Update TaskList class with print statements

rayyan35p committed:
Integrate new task classes into main code

rayyan35p committed:
Remove redundant line

rayyan35p committed:
Update tests

rayyan35p committed:
Add DukeException

rayyan35p committed:
Modify main code to catch exceptions

rayyan35p committed:
Add deleteTask method to TaskList class

rayyan35p committed:
Modify main code to allow chatbot to delete tasks from list

rayyan35p committed:
Add code to create a folder and file to store tasks

rayyan35p committed:
Update .gitignore to ignore files in data folder

rayyan35p committed:
Update task classes with methods needed for I/O

rayyan35p committed:
Update TaskList class to write tasks to a file

rayyan35p committed:
Integrate file writing operations into main

rayyan35p committed:
Modify main code to read previously stored tasks from a file

rayyan35p committed:
Merge branch 'branch-Level-7'

rayyan35p committed:
Fix bug where directory to store tasks is not made

rayyan35p committed:
Add LocalDateTime functionality to task classes

rayyan35p committed:
Update LocalDateTime functionality in task classes

rayyan35p committed:
Merge branch 'branch-Level-8'

rayyan35p committed:
Add Storage class for handling file operations

rayyan35p committed:
Integrate Storage class in main code

rayyan35p committed:
Add Parser Class

rayyan35p committed:
Fix formatting

rayyan35p committed:
Add new methods to TaskList class for executing commands

rayyan35p committed:
Integrate Parser class into main code

rayyan35p committed:
Add Ui class

rayyan35p committed:
Integrate Ui class into main code

rayyan35p committed:
Update input.txt with correct datetime format

rayyan35p committed:
Add new method to Ui class

rayyan35p committed:
Implement Ui.bye() for bye command in main code

rayyan35p committed:
Fix formatting

rayyan35p committed:
Merge branch 'branch-MoreOOP'

rayyan35p committed:
Put classes into packages

rayyan35p committed:
Modify expected output to match new input text for tests

rayyan35p committed:
Merge branch 'branch-Packages'

rayyan35p committed:
Merge branch 'add-gradle-support'

rayyan35p committed:
Set mainClassName to be Duke

rayyan35p committed:
Add new method open() for reading file into TaskList

rayyan35p committed:
Add data folder in text-ui-text to .gitignore

rayyan35p committed:
Update error message for FileNotFoundException in open()

rayyan35p committed:
Add JUnit test for Task classes

rayyan35p committed:
Merge branch 'branch-JUnit'

rayyan35p committed:
Add JavaDoc

rayyan35p committed:
Fix formatting

rayyan35p committed:
Add print method to Ui

rayyan35p committed:
Modify methods in TaskList

rayyan35p committed:
Add find feature to main code with its relevant methods in the classes

rayyan35p committed:
Merge branch 'branch-A-CodingStandard'

rayyan35p committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/data/task/TaskList.java

rayyan35p committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/data/exception/DukeException.java
#	src/main/java/data/task/TaskList.java
#	src/main/java/ui/Ui.java

rayyan35p committed:
Fix variable names

dishenggg commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,38 @@
+        Storage storage = new Storage();
+        TaskList list = new TaskList(storage.getFile());
+        list.open();
+        Parser parser = new Parser();
```
Maybe you can consider initializing all your variables in a section on top and leaving a line before running any logic?

rayyan35p replied:
Yeah that would be neater, thanks!

dishenggg commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,114 @@
+            }
+        case UNMARK:
+            if (inputSplit.length == 1) {
+                throw new DukeException("Input the task number");
```
Perhaps you can consider using a named constant for `"Input the task number"` and `"Invalid command"`?

rayyan35p replied:
Yep ok makes sense

dishenggg commented on src/main/java/data/task/TaskList.java:
```diff
@@ -0,0 +1,296 @@
+     * @param input The input string for the command.
+     * @throws DukeException If an EventTask is instantiated with invalid dates.
+     */
+    public void executeCommand(Command command, String input) throws DukeException {
```
I like that you delegated the logic of each command to their own function. It reduced the overall length of the methods

dishenggg commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,38 @@
+
+        String input = ui.getFirstInput();
+
+        while (true) {
```
Well structured code, no deep nesting 👍

dishenggg commented on src/main/java/data/task/TaskList.java:
```diff
@@ -0,0 +1,296 @@
+            input = input.substring(6);
+            inputArr = input.split(" /from ");
+            String description = inputArr[0];
+            inputArr = inputArr[1].split(" /to ");
```
Maybe you can consider using different variables names instead of overwriting them?

rayyan35p replied:
Will do, thanks!

dishenggg reviewed (commented):
LGTM! Good code quality throughout

hjoneweek reviewed (commented):
It seems like there weren't any coding standard violations (that I could find) in the program. I found your code easy to read as well. Looks good to me!

rayyan35p committed:
Add JavaFX dependencies to build.gradle

rayyan35p committed:
Move all files into bob package

rayyan35p committed:
Add GUI elements

rayyan35p committed:
Add changes to GUI

rayyan35p committed:
Integrate commands into GUI

rayyan35p committed:
Modify GUI to close on bye command

rayyan35p committed:
Merge branch 'branch-GUI'

rayyan35p committed:
Add checkstyle

rayyan35p committed:
Fix checkstyle issues

rayyan35p committed:
Remove deep nesting

rayyan35p committed:
Add Command classes

rayyan35p committed:
Move logic of commands into individual Command classes

Execution of commands is being done by TaskList class instead of
command classes.

It may be difficult to ascertain any bugs with Command logic or TaskList
logic if they are not separated.

The logic of commands being executed will be handled by the command
classes instead of the TaskList class so TaskList will only have methods
relevant to the TaskList.

With proper separation of behaviors into relevant classes, the code will
be clearer.

rayyan35p committed:
Add Javadocs

rayyan35p committed:
Remove redundant lines

rayyan35p committed:
Add assertions

rayyan35p committed:
Merge pull request #2 from rayyan35p/branch-A-Assertions

Add assertions

rayyan35p committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/bob/data/task/TaskList.java
#	src/main/java/bob/parser/Parser.java

rayyan35p committed:
Merge pull request #3 from rayyan35p/branch-A-CodeQuality

Merge Code Quality Branch

rayyan35p committed:
Add ability to update tasks in the list

Not able to update the details of tasks.

Users will have to delete tasks and re-add them back with the correct
details in order to "update" them.

Let's make a new UpdateCommand to facilitate updating tasks.

Updating tasks will make the user experience more convenient.

rayyan35p committed:
Merge branch 'branch-C-Update'

rayyan35p committed:
Add more unit tests

rayyan35p committed:
Merge branch 'branch-A-MoreTesting'

rayyan35p committed:
Merge branch 'master' into branch-A-MoreErrorHandling

rayyan35p committed:
Fix bugs that caused some tests to fail

rayyan35p committed:
Fix formatting

rayyan35p committed:
Fix bug with invalid dates

rayyan35p committed:
Refactor parse into separate parse methods for each command

rayyan35p committed:
Merge branch 'branch-A-MoreErrorHandling'

rayyan35p committed:
Fix GUI not showing long dialogs

rayyan35p committed:
Modify GUI to send user input with enter key

rayyan35p committed:
Merge branch 'branch-A-BetterGui'

rayyan35p committed:
Add canWeFixIt()

rayyan35p committed:
Add title to GUI window

rayyan35p committed:
Update README

rayyan35p committed:
Fix README formatting

rayyan35p committed:
Fix README formatting

rayyan35p committed:
Fix formatting in README

rayyan35p committed:
Fix minor issues
